cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceEnsurer.java

```diff
@@ -67,7 +69,14 @@ protected void doStop() {
           }
         };
       }
-    }, RetryStrategies.exponentialDelay(200, 5000, TimeUnit.MILLISECONDS));
+    }, RetryStrategies.timeLimit(10, TimeUnit.MINUTES,
+      RetryStrategies.exponentialDelay(200, 5000, TimeUnit.MILLISECONDS)));
+    serviceDelegate.addListener(new ServiceListenerAdapter() {
+      @Override
+      public void failed(State from, Throwable failure) {
+        LOG.error("Received failure from state '{}'.", from, failure);
```

Comment:
This was for logging when it times out. Removed the time limit, so removing this now.